### PR TITLE
chore: Update doc-examples tag to 3.0

### PR DESCRIPTION
**Pull request by mheap**
#### What this PR does / why we need it:

Updates the Kong Gateway tag to 3.0

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] PR is based off the current tip of the `main` branch.
- [ ] Changes are documented under the "Unreleased" header in CHANGELOG.md <- Not required as it's an example values file
- [x] New or modified sections of values.yaml are documented in the README.md
- [x] Commits follow the [Kong commit message guidelines](https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#commit-message-format)

### CODE REVIEW

File: charts/kong/example-values/doc-examples/quickstart-enterprise-licensed-aio.yaml [hunk 1]
```diff
@@ -151,7 +151,7 @@ extraLabels:
   konghq.com/component: quickstart
 image:
   repository: kong/kong-gateway
-  tag: "2.8"
+  tag: "3.0"
```
Comment (pmalek) [inline charts/kong/example-values/doc-examples/quickstart-enterprise-licensed-aio.yaml:154]: 🤔 Which registry + repo actually do we mean here? This doesn't seem to exist under: https://hub.docker.com/kong/kong-gateway

Comment (mheap) [reply charts/kong/example-values/doc-examples/quickstart-enterprise-licensed-aio.yaml:154]: image

It exists for me

Comment (pmalek) [reply charts/kong/example-values/doc-examples/quickstart-enterprise-licensed-aio.yaml:154]: image

😿

Comment (mheap) [reply charts/kong/example-values/doc-examples/quickstart-enterprise-licensed-aio.yaml:154]: Ah, it's a public image but can't be listed publicly. `docker pull kong/kong-gateway:3.0` should work fine for you

Comment (pmalek) [reply charts/kong/example-values/doc-examples/quickstart-enterprise-licensed-aio.yaml:154]: 👍

**Comment by rainest** [comment]
https://github.com/Kong/charts/pull/666/files will hit all of these but isn't going to proceed until the `docker.io/kong` image and controller 2.6 image are out. KIC 2.5 has compatibility issues with 3.x so we do not want to bump the versions piecemeal.

**Pull request closed without merging**